sources/ClangSharp.PInvokeGenerator/CSharp/CSharpOutputBuilder.VisitDecl.cs

```diff
@@ -372,6 +372,9 @@ public void BeginFunctionOrDelegate(in FunctionOrDelegateDesc desc, ref bool isM
                 Write(desc.ParentName);
                 Write('.');
                 Write(desc.EscapedName);
+                Write('(');
+                Write(string.Join(", ", desc.ParameterTypes));
+                Write(')');
```

Comment:
nit: It would be beneficial to only do this when there are potential conflicts, it's a lot of extra text that's unnecessary in most cases (particularly for regular C exports, since its really only applicable to C++ exports).

There's some existing helper functions that determine if multiple members with the same name exist that can likely be used.

Reply:
fair, do you know where those functions live off hand?

Reply:
@curin sorry, thought I had already responded to this.

You likely want to reuse a similar approach to `FixupNameForMultipleHits`: https://source.clangsharp.dev/#ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitDecl.cs,dbbe5069d6d4e556,references

It might involve tweaking `GetOverloadIndex` slightly to allow disambiguating "no overloads" (-1) from "some overloads" (>= 0), where today we just track a raw index instead not how many overloads existed.

Reply:
I ended up adding a function just to get the overload count based on `GetOverloadIndex`. The only difference is that I doesn't cache the count like indicies do. I can add that if you want, but wasn't going to add the memory overhead unless you thought it was a good idea.